### PR TITLE
Logging samples with Google.Logging.V2

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: logging/api/QuickStart/QuickStart.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using Google.Api;
+
+namespace GoogleCloudSamples
+{
+    public class QuickStart
+    {
+        static void Main(string[] args)
+        {
+            // Your Google Cloud Platform project ID.
+            string projectId = "YOUR-PROJECT-ID";
```
Comment (SurferJeffAtGoogle) [inline logging/api/QuickStart/QuickStart.cs]: Let's add a check detecting YOUR-PROJECT-ID and printing a message that says you need to edit the code.

File: logging/api/LoggingTest/LoggingTest.cs [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                output = consoleApp.StandardOutput.ReadToEnd();
+                consoleApp.WaitForExit();
+                return output;
+            }
+
+            [Fact]
+            public void TestQuickStartConsoleApp()
+            {
+                string output;
+                string filePath = "..\\..\\..\\QuickStart\\bin\\Debug\\QuickStart.exe";
```
Comment (SurferJeffAtGoogle) [inline logging/api/LoggingTest/LoggingTest.cs]: Use an @"string" and then you don't need to escape the slashes.

File: logging/api/Program.cs [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            var results = client.ListLogEntries(projectIds, logName, "timestamp desc");
+            foreach (var row in results)
+            {
+                if (row != null && !String.IsNullOrEmpty(row.TextPayload.Trim()))
+                {
+                    _out.WriteLine($"{row.TextPayload.Trim()}");
+                }
+                else
+                {
+                    results.GetEnumerator().Dispose();
```
Comment (SurferJeffAtGoogle) [inline logging/api/Program.cs]: Instead, let's change line 97 to 

using (var results = client.ListLogEntries(projectIds, logName, "timestamp desc"))
{

File: logging/api/LoggingTest/LoggingTest.cs [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+        protected static void AssertSucceeded(ConsoleOutput output)
+        {
+            Assert.True(0 == output.ExitCode,
+                $"Exit code: {output.ExitCode}\n{output.Stdout}");
+        }
+
+        public class LoggingTest : BaseTest, IDisposable
+        {
+            public void Dispose() { }
```
Comment (SurferJeffAtGoogle) [inline logging/api/LoggingTest/LoggingTest.cs]: If Dispose() is empty, then we can just drop it and stop implementing IDisposable.

File: logging/api/QuickStart/QuickStart.cs (hunk 1 above)
Comment (jsimonweb) [inline logging/api/QuickStart/QuickStart.cs]: Done.

File: logging/api/LoggingTest/LoggingTest.cs (hunk 4 above)
Comment (jsimonweb) [inline logging/api/LoggingTest/LoggingTest.cs]: Done.

File: logging/api/Program.cs (hunk 3 above)
Comment (SurferJeffAtGoogle) [inline logging/api/Program.cs]: I downloaded the code and tried running it, and it blocks forever even with the work-around.  :-(

File: logging/api/LoggingTest/LoggingTest.cs [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+                string message = "Example log entry.";
+                // Try creating a log entry.
+                var created = Run("create-log-entry", logId, message);
+                AssertSucceeded(created);
+                // Pause for 5 seconds before trying to get newly added log entry.
+                Thread.Sleep(5000);
+                // Retrieve the log entry just added, using the logId as a filter.
+                var results = Run("list-log-entries", logId);
+                // Confirm returned log entry contains expected value.
+                Assert.Contains(message, results.Stdout);
```
Comment (SurferJeffAtGoogle) [inline logging/api/LoggingTest/LoggingTest.cs]: If the assertion fails, then the Run("delete-log") statement below will never run, and the test will leak a log.

This pattern repeats in the logs below.  Adding try .. finally .. to every test would be tedious.  Instead, you could add a list of garbage as a member variable for the class, and then implement Dispose() to delete the garbage.

Comment (jsimonweb) [inline logging/api/LoggingTest/LoggingTest.cs]: Done.

File: logging/api/LoggingTest/LoggingTest.cs [hunk 6]
```diff
@@ -64,13 +68,50 @@ protected static void AssertSucceeded(ConsoleOutput output)
...
+            {
+                try
+                {
+                    // Delete all logs created from running the tests.
+                    foreach (string log in _logsToDelete)
+                    {
+                        Run("delete-log", log);
+                    }
+                }
+                catch (RpcException ex)
```
Comment (SurferJeffAtGoogle) [inline logging/api/LoggingTest/LoggingTest.cs]:
```
catch (RpcException ex) when (x.Status.StatusCode == StatusCode.NotFound) {}
```

File: logging/api/LoggingTest/LoggingTest.cs [hunk 7]
```diff
@@ -64,13 +68,50 @@ protected static void AssertSucceeded(ConsoleOutput output)
...
+                }
+                try
+                {
+                    // Delete all the log sinks created from running the tests.
+                    foreach (string sink in _sinksToDelete)
+                    {
+                        Run("delete-sink", sink);
+                    }
+                }
+                catch (RpcException ex)
```
Comment (SurferJeffAtGoogle) [inline logging/api/LoggingTest/LoggingTest.cs]: catch(...) when (...) {}

File: logging/api/LoggingTest/LoggingTest.cs (hunk 6 above)
Comment (jsimonweb) [inline logging/api/LoggingTest/LoggingTest.cs]: Done.

File: logging/api/LoggingTest/LoggingTest.cs (hunk 7 above)
Comment (jsimonweb) [inline logging/api/LoggingTest/LoggingTest.cs]: Done

**Pull request merged**